client/src/components/addName.js

```diff
@@ -70,18 +70,19 @@ function AddName() {
         //unique constraint error(name already exists)
         if (response.status === 422) {
           setUniqueError(true);
+          return Promise.reject(response);
```

Comment:
Setting state occurs asynchronously in a promise. Because of this, checking the `uniqueError` state in the following `.then()` doesn't yield the true value. This solution rejects the promise here so we don't attempt to parse the json data.

The downside is we don't parse the server error message at all, but since we're only expecting uniqueness errors this solution is safe for now. It may yield confusing errors in the future if we enhance this form at all though.

Reply:
Nice catch.  Very interesting.  Thanks for the in-depth explanation.